### DISCUSSION TITLE
cymk offset effect.

### DISCUSSION THREAD

**Discussion by artnose** [ImageMagick/ImageMagick#3770] [Digital Image Processing]
I'm trying to reproduce this effect   https://i.imgur.com/kRRwiiB.jpeg

Has anyone done this?   From a normal image, I'd lighten the background, 
then shift the c,y,m,k layers a bit.

**Comment by snibgo** [comment]
Yes, you could do that. You could separate channels, and shift two channels eg with "-roll", and combine them.

If your input is a black object on a white background, then shifting RGB channels by more than the object size will give that effect.

If you want more help, please say what version of IM, on what platform, and supply an input and required output.

**Comment by artnose** [comment]
Yeah, I'm using  im 7 on command line linux. 

I'd guess something like this  for Yellow, but it doesn't seem to shift yellow, just black & white
convert input.jpg -colorspace CMYK \( -clone 0 -channel Y -roll -20  \) -composite output.jpg

I'm just not sure how to separate and combine the channels again.

**Comment by snibgo** [comment]
Let's make a test image with a black object on white background:
```
magick \
  -size 200x200 xc:White \
  -fill Black -draw "circle 100,100,115,100" \
  x.png
```
![x](https://user-images.githubusercontent.com/33812368/120355197-0025f700-c2fb-11eb-8397-a0955d938abc.png)

Separate, shift the R and B components, and combine:
```
magick \
  x.png \
  -set colorspace sRGB \
  -channel 0,1,2 \
  -separate \
  \( -clone 0 -roll +20+0 \) -swap 0,-1 +delete \
  \( -clone 1 -roll +40+0 \) -swap 1,-1 +delete \
  -combine \
  x1.png
```
![x1](https://user-images.githubusercontent.com/33812368/120355245-07e59b80-c2fb-11eb-8685-920c0769a361.png)

**Comment by artnose** [comment]
perfect!  Thank you so much!

**Comment by hubagaspar91** [comment]
Hey! Came across this thread, and there is something very similar that I'd like to do, but with RGB channels instead of CMYK. How do I modify the above command for that?

**Comment by snibgo** [comment]
How about:
```
magick \
  -size 200x200 xc:Black \
  -fill White -draw "circle 100,100,115,100" \
  x.png

magick \
  x.png \
  -set colorspace sRGB \
  -channel 0,1,2 \
  -separate \
  \( -clone 0 -roll +20+0 \) -swap 0,-1 +delete \
  \( -clone 1 -roll +40+0 \) -swap 1,-1 +delete \
  -combine \
  x2.png
```
![x2](https://user-images.githubusercontent.com/33812368/138143844-6bc0a081-7288-4e50-aaff-0362ab066aa9.png)
